### DISCUSSION TITLE
Seeking Assistance for Floating Element Collision Detection and Dynamic Positioning Update

### DISCUSSION THREAD

**Discussion by totorofly** [floating-ui/floating-ui#2523] [Ideas]
Hi,

I've been using Floating-UI and greatly appreciate its powerful ability to handle the dynamic positioning update of floating elements along with its robust element collision detection. I've been typically dealing with situational updates of the floating elements, such as when an element originally placed at the bottom is automatically adjusted to the top placement upon detecting a collision with the bottom visible area.

However, I've encountered a scenario that is converse to this typical situation; I hope to scroll the viewport (i.e., to scroll down, similar to window.scrollTop) rather than adjust the positioning of the floating element when a collision with the bottom is detected. This way, the floating element is fully displayed by adjusting the viewport to slide-down rather than changing its bottom placement. I am seeking to know if Floating-UI currently has any existing parameters or APIs that support this particular feature. 

Thank you and looking forward to your valuable suggestions and recommendations!

**Comment by atomiks** [comment]
I'm not sure if I understand what you're asking. Can you try to rephrase it?

If I had to guess, maybe you want flipping to happen only when the floating element first opens and gets positioned, and not when scrolling afterwards? See this comment/sandbox for a solution: https://github.com/floating-ui/floating-ui/issues/2222#issuecomment-1464968480

**Reply by atomiks** [reply]
Alternatively, maybe you're asking about this: https://github.com/floating-ui/floating-ui/issues/1871#issuecomment-1229364230

**Reply by totorofly** [reply]
After referring to the previous answers, I think I've successfully written a solution that suits my scenario. The code is as follows: 

```
React.useLayoutEffect(() => {
  if (isOpen) {
    requestAnimationFrame(() => {
      const container = document.documentElement; // Represents the distance from the top of the document to the top of the current visible viewport
      const windowHeight = window.innerHeight; // Represents the height of the current viewport
      const element = refs.floating.current as HTMLElement;
      const elementClientRectBottom = element.getBoundingClientRect().bottom; // Represents the distance between the bottom edge of the element and the top edge of the viewport in pixels
      const overflowBottom = elementClientRectBottom - windowHeight; // Represents the distance between the bottom edge of the element and the bottom edge of the viewport in pixels
      
      let scrollDistance;
      const elementHeight = element.getBoundingClientRect().height;

      if (elementClientRectBottom > windowHeight) {
        // If the bottom of the element exceeds the bottom of the window, scrolling is needed
        if (elementHeight > windowHeight) {
          // If the height of the element exceeds the height of the window, scroll to the top of the element
          scrollDistance = container.scrollTop + element.getBoundingClientRect().top;
        } else {
          // Otherwise, scroll to the bottom of the element
          scrollDistance = container.scrollTop + (overflowBottom > 0 ? overflowBottom : 0);
        }

        window.scrollTo({
          top: scrollDistance,
          behavior: "smooth",
        });
      }
    });
  }
}, [isOpen, refs]);

```


Thank you again for your help.

**Comment by totorofly** [comment]
> Alternatively, maybe you're asking about this: [#1871 (comment)](https://github.com/floating-ui/floating-ui/issues/1871#issuecomment-1229364230)

Thank you very much for your answer! My issue seems to be very similar to the situation in #1871. I will try the solution mentioned in #1871 first. Thanks!
